### PR TITLE
feat(api): allow redirects with a returnTo param

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
~~This should not touch production code at all.~~  

The `session.returnTo` was only used in dev login, so I got rid of it.

@moT01 if you've got a chance, could you take a look?

Related: https://github.com/freeCodeCamp/freeCodeCamp/pull/40160

### CODE REVIEW

File: api-server/server/boot/authentication.js [hunk 1]
```diff
@@ -37,8 +37,14 @@ module.exports = function enableAuthentication(app) {
...
-  const saveAuthCookies = saveResponseAuthCookies();
-  const loginSuccessRedirect = loginRedirect();
+  const devSaveAuthCookies = devSaveResponseAuthCookies();
+  const devLoginSuccessRedirect = devLoginRedirect();
+  const addRedirect = (req, res, next) => {
+    if (req && req.query && req.query.returnTo) {
+      req.query.returnTo = `${homeLocation}/${req.query.returnTo}`;
+    }
+    return next();
+  };
```
Comment (raisedadead) [inline api-server/server/boot/authentication.js:47]: This part of the logic affects what gets executed here: https://github.com/freeCodeCamp/freeCodeCamp/pull/40161/files#diff-1f182080313f7e6aae86f53b770161759a3008abae3e3bdba965fadfd7ed4068R118 - unless I am totally missing on something.

Comment (ojeytonwilliams) [reply api-server/server/boot/authentication.js:47]: Just to be transparent (we discussed this elsewhere):

> Unlike a session, a query string can be added by anyone including client?

I can't see what the diff is referring to, but this is just a refactor.  The danger arises if we allow arbitrary directs, but here we control `homeLocation`.  But, yes, the `returnTo` string could be anything, so any change that allows redirection to other domains other than www.freecodecamp.org will need to be implemented carefully.

Comment (raisedadead) [reply api-server/server/boot/authentication.js:47]: Yes - that is correct. I fixed my comment above to point to the code where it can get picked up.

**Review by moT01 — COMMENTED** [review]
I dunno - I looked through this and tested it out pretty thoroughly - I didn't see any problems, but I feel like I may need a little walk through of it to fully understand it.

**Review by raisedadead — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
Thanks @moT01. Any sanity checking helps.

**Pull request merged**